UE/Tests/Application/SmsDbTestSuite.cpp

```diff
@@ -26,4 +26,29 @@ TEST_F(SmsDbTestSuite, shallAddReceivedSms)
     ASSERT_EQ(smsState, SmsState::NotViewed);
 }
 
+TEST_F(SmsDbTestSuite, shallgetSms)
+{
+    const Sms sms{phoneNumber, "example text"};
+    objectUnderTest.addReceivedSms(sms);
+
+    const auto& smsFromDb = objectUnderTest.getSms(0);
+    ASSERT_EQ(smsFromDb, sms);
+}
+
+TEST_F(SmsDbTestSuite, shallUpdateSmsState)
+{
+    const Sms sms{phoneNumber, "example text"};
+    objectUnderTest.addReceivedSms(sms);
+
+    const auto& messagesBeforeUpdate = objectUnderTest.getSmsMessages();
+    auto [sms1, smsStateBeforeUpdate] = messagesBeforeUpdate[0];
+    ASSERT_EQ(smsStateBeforeUpdate, SmsState::NotViewed);
+
+    objectUnderTest.updateSmsState(0);
+
+    const auto& messagesAfterUpdate = objectUnderTest.getSmsMessages();
+    auto [sms2, smsStateAfterUpdate] = messagesAfterUpdate[0];
```

Comment:
unnecessary references, just use the previous ones

Reply:
which are unnecessary? update is going on the smsDb, to check if it is updated I need to retrieve messages again